test/e2e_env/kubernetes/healthcheck/virtual_probes.go

```diff
@@ -37,26 +32,18 @@ func VirtualProbes() {
 		Expect(err).To(Succeed())
 	})
 
-	PollPodsReady := func() error {
-		pods, err := k8s.ListPodsE(env.Cluster.GetTesting(), env.Cluster.GetKubectlOptions(namespace),
-			metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", name)})
-		if err != nil {
-			return err
-		}
-		for _, p := range pods {
-			err := k8s.WaitUntilPodAvailableE(env.Cluster.GetTesting(), env.Cluster.GetKubectlOptions(namespace), p.GetName(), 0, 0)
-			if err != nil {
-				return err
-			}
-		}
-		return nil
-	}
-
 	It("should deploy test-server with probes", func() {
 		// Sample pod readiness to ensure they stay ready to at least 10sec.
-		for i := 0; i < 10; i++ {
-			time.Sleep(time.Second)
-			Expect(PollPodsReady()).To(Succeed())
-		}
+		Eventually(func(g Gomega) {
+			pods, err := PodsAvailable(env.Cluster, name, namespace)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(pods).To(Equal(1))
+		}, "30s", "1s").Should(Succeed())
+
+		Consistently(func(g Gomega) {
+			pods, err := PodsAvailable(env.Cluster, name, namespace)
```

Comment:
Maybe something to add upstream to terratest?

Reply:
let me add this to my todo list. I'll try to contribute when I have time

Reply:
Yeah definitely not for this PR.